### ISSUE TITLE
Swagger 2.0 Support

### ISSUE THREAD

**Issue by wedtm** [ruby-grape/grape-swagger#219]
Is this possible yet?

**Comment by calfzhou** [comment]
really need 2.0 support, swagger 2.0's authorization supports oauth2 4 flows, but only 2 flows for version 1.2.

**Comment by dblock** [comment]
I'd really like to see https://github.com/tim-vandecasteele/grape-swagger/issues/137, in which we rewrite grape-swagger using a proper swagger DSL library instead of keeping at hacking on stuff. That brings us to Swagger 2.0. And anyone who does it will be my personal hero.

**Comment by tbeauvais** [comment]
:+1:

**Comment by danielkummer** [comment]
:+1:

**Comment by florrain** [comment]
Any news about that?

:+1: 
:+1: 
:+1:

**Comment by yairgo** [comment]
Has anyone thought about this at all yet? Is there a path going forward that we can get started and maybe have multiple people work on it at once?

**Comment by dblock** [comment]
Doesn't look like anyone has done anything about this, the first one wins!

**Comment by birbird** [comment]
How about this 
https://github.com/Gild/ruby-swagger

**Comment by dmitry** [comment]
As I understand it's not generating documentation on the fly. But it has support for grape: https://github.com/Gild/ruby-swagger/tree/master/lib/ruby-swagger/grape

**Comment by openmosix** [comment]
@dmitry correct, there is a rake task - extracting the data from grape into a YML format - this allows tech editors to write all the doc outside the annotations. The documentation can be compiled into its final swagger.json form - this allow us to host the documentation on CDN rather than from the API server.

**Comment by aceunreal** [comment]
:+1:

**Comment by korstiaan** [comment]
:+1:

**Comment by wagner** [comment]
:+1:

**Comment by Micka33** [comment]
would be awesome.

**Comment by johnhamelink** [comment]
:+1:

**Comment by herberzt** [comment]
Is the only current strategy to get 2.0 Swagger compliance still https://github.com/Gild/ruby-swagger? Are there any plans to add 2.0 support to this gem?

**Comment by dblock** [comment]
Nobody is actively working on this @herberzt. Maybe you will?

**Comment by edbo** [comment]
Can one of the original maintainers outline how someone would go about this task?

Right now, for someone new using this project, it's not necessarily obvious how to go about starting to do this.

@dblock I'm not suggesting you take this task up but it's currently a bit daunting for someone that hasn't maintained this repo before to take up and it's definitely something I'd like to see happen.

**Comment by dblock** [comment]
I am here to help. What I need is a passing build in a pull request, and I can definitely carry it forward after it - https://github.com/ruby-grape/grape-swagger/pull/311 was very promising.

**Comment by dblock** [comment]
https://github.com/ruby-grape/grape-swagger/pull/311 has been merged onto a branch called swagger-2.0. Looking forward to more PRs to move that forward!

**Comment by pswenson** [comment]
where do things stand wtih the 2.0 branch:  https://github.com/LeFnord/grape-swagger ?

We are trying it out, but it's unclear where things stand.

**Comment by XBeg9** [comment]
:+1:

**Comment by dblock** [comment]
So we have a 2.0 branch. Who wants to take ownership, get that merged onto master, released, etc.? What remains to get done? @LeFnord maybe?

**Comment by LeFnord** [comment]
Hi guys … I would prefer to create a new grape-swagger-2 repo under the ruby-grape project, so much more people can contribute to it and it feels more on the right place.
An side effect of this would be, that we can start a discussion on decisions, cause I don't want to decide all alone, think this would not be a good idea :wink:  … there is something to decide, e.g. how to handle grape specific behavior → `exactly_one_of`, `mutually_exclusive` to name only some …

**Comment by dblock** [comment]
I am not in love with the idea of two gems for different versions of the spec. I think people just want to upgrade to the swagger 2.0 spec soon and call it a day, do you see a reason one wouldn't want to?

**Comment by LeFnord** [comment]
@dblock I hope and I wish … then it would be best as it is for the moment, or?
or did you prefer to use my fork for that, this is also ok for me …

**Comment by dblock** [comment]
I'd like to do the following:

1. Release 0.10.5.
2. Branch 0.10.5 into 0.10.x, continue releasing bug fixes only in a 0.10.x series compatible with 1.5
3. Merge swagger-2.0 onto master, release 0.11.0 with only Swagger 2.0 support.
4. Move forward.

Thoughts? 

If you agree with this @LeFnord I want you to take care of (3) and (4). I've not been actively developing this library I have been just cheerleading the effort here and would love for more people to take a very active role. Who can be better than the person who wrote the majority of 2.0 support for doing that? :)

**Comment by LeFnord** [comment]
Hi @dblock mmh ok, but for (1)..(3) I need your advise, cause I thing I haven't the rights to do so, and yeap it also would be the first time :wink: 

point (3) would prefer to start with tag 0.20.x, so the other branch has enough version space for changes,
and it indicates a little more breaking changes (ok, doing it right SemVer, we have to use 1.0.0)
point (4) definitely

**Comment by dblock** [comment]
@LeFnord I invited you the swagger-maintainers org. I am happy to help with whatever you need. 

You should PR yourself under grape-swagger in https://github.com/ruby-grape/ruby-grape.github.io/blob/da76718ff723d8e6e6a3f45972045ce133220842/team/index.md.

Drop me an email at dblock[at]dblock[dot]org with your Rubygems e-mail address for access to that, since you're making the next release.

We should introduce proper semantic versioning, whatever that means, trust your judgement on which version this should be. Also cc: @tim-vandecasteele and @CraigCottingham wrt ^^^ who are the other two maintainers on this project.

Finally, thank you for spearheading this.

**Comment by florrain** [comment]
Looks like you're working out a solution, that's awesome!

Do you have a rough idea in terms of release timeline (either in days/weeks/months)? Like a first beta release that the public could test?

**Comment by LeFnord** [comment]
@florrain the release is in discussion …
but for first one could use the [swagger-2.0 branch](https://github.com/ruby-grape/grape-swagger/tree/swagger-2.0) and give feedback,
or much more preferred commit fixes :wink:

**Comment by florrain** [comment]
Got it, I'm seeing exceptions using that branch. Should I create an issue on this repo or do you have a discussion channel I can hop in?

**Comment by LeFnord** [comment]
I personally prefer issues … or some slack lobby

**Comment by dblock** [comment]
(Maybe we should move the mailing list to slack, I wonder ...) I opened https://github.com/ruby-grape/ruby-grape.github.io/issues/29 if anyone feels strongly comment there.

**Comment by florrain** [comment]
Created an issue for you @LeFnord. Down for Slack but maybe as a complementary tool for discussions between maintainers and direct/quick support, not a replacement of the mailing list even though I'm personally not a fan of the concept.

**Comment by LeFnord** [comment]
yeap, definitely not a replacement of a mailing list

**Comment by dblock** [comment]
FYI I created issue labels for swagger 1.2 and 2.0, should help out with sorting.
